packages/repack/src/webpack/plugins/OutputPlugin.ts

```diff
@@ -213,205 +218,202 @@ export class OutputPlugin implements WebpackPlugin {
-    let entryChunk: webpack.Chunk | undefined;
-    const entryChunkName = this.config.entryName ?? 'main';
-    const localChunks: webpack.Chunk[] = [];
-    const remoteChunks: webpack.Chunk[] = [];
-    const auxiliaryAssets: Set<string> = new Set();
-
-    compiler.hooks.compilation.tap('RepackOutputPlugin', (compilation) => {
-      compilation.hooks.processAssets.tap(
-        {
-          name: 'RepackOutputPlugin',
-          stage: webpack.Compilation.PROCESS_ASSETS_STAGE_ADDITIONS,
-        },
-        () => {
-          entryGroup = compilation.chunkGroups.find((group) =>
-            group.isInitial()
-          );
-          entryChunk = entryGroup?.chunks.find(
-            (chunk) => chunk.name === entryChunkName
-          );
-          const sharedChunks = new Set<webpack.Chunk>();
-
-          for (const chunk of compilation.chunks) {
-            // Do not process shared chunks right now.
-            if (sharedChunks.has(chunk)) {
-              continue;
-            }
-
-            [...chunk.getAllInitialChunks()]
-              .filter((sharedChunk) => sharedChunk !== chunk)
-              .forEach((sharedChunk) => {
-                sharedChunks.add(sharedChunk);
-              });
+    const getRelatedSourceMap = (chunk: webpack.StatsChunk) => {
+      return chunk.auxiliaryFiles?.find((file) => /\.map$/.test(file));
+    };
 
-            // Entry chunk
-            if (entryChunk && entryChunk === chunk) {
-              localChunks.push(chunk);
-            } else if (isLocalChunk(chunk.name ?? chunk.id?.toString())) {
-              localChunks.push(chunk);
-            } else {
-              remoteChunks.push(chunk);
-            }
-          }
+    const getAllInitialChunks = (
+      chunk: webpack.StatsChunk,
+      chunks: Map<string | number, webpack.StatsChunk>
+    ): Array<webpack.StatsChunk> => {
+      if (!chunk.parents?.length) return [chunk];
+      return chunk.parents.flatMap((parent) => {
+        return getAllInitialChunks(chunks.get(parent)!, chunks);
+      });
+    };
 
-          // Process shared chunks to add them either as local or remote chunk.
-          for (const sharedChunk of sharedChunks) {
-            const isUsedByLocalChunk = localChunks.some((localChunk) => {
-              return [...localChunk.getAllInitialChunks()].includes(
-                sharedChunk
-              );
-            });
-            if (
-              isUsedByLocalChunk ||
-              isLocalChunk(sharedChunk.name ?? sharedChunk.id?.toString())
-            ) {
-              localChunks.push(sharedChunk);
-            } else {
-              remoteChunks.push(sharedChunk);
-            }
-          }
+    compiler.hooks.done.tapPromise('RepackOutputPlugin', async (stats) => {
+      const compilationStats = stats.toJson({
+        all: false,
+        assets: true,
+        chunks: true,
+        chunkRelations: true,
+        ids: true,
+      });
+      const chunks = new Map(
```

Comment:
NIT: consider renaming this to sth like `allCompilationChunks` for readability – this map is used multiple times and I found myself wondering whether it was the "all chunks map" or some more scoped temp variable when building specific (local/remote/shared) chunks lists